specs/users.table

```diff
@@ -9,7 +9,7 @@ schema([
     Column("description", TEXT, "Optional user description"),
     Column("directory", TEXT, "User's home directory"),
     Column("shell", TEXT, "User's configured default shell"),
-    Column("uuid", TEXT, "User's UUID (Apple) or SID (Windows)"),
+    Column("uuid", TEXT, "User's UUID (Apple) or SID (Windows)", index=True),
```

Comment:
This is not indexed on darwin

Reply:
Indeed there is no `uuid` index implemented on macOS, or Linux for that matter. Although, this apparently has no impact to functionality, just as the Windows `users` implementation not implementing index on the `uid` column prior to this PR.

The `uuid` exists on macOS and is populated. But there are no other tables on macOS currently that use user UUIDs except `disk_encryption` (which currently leaves that column blank anyway, perhaps that's a bug), instead they all deal in the short format UID.

On master branch, here is the output of `./build/osquery/osqueryi --planner "select * from managed_policies JOIN users USING (uuid)"`:

```osquery planner: xBestIndex Evaluating constraints for table: managed_policies [index=0 column=1 term=0 usable=1]
osquery planner: xBestIndex Recording constraint set for table: managed_policies [cost=1000000.000000 size=0 idx=0]
osquery planner: xBestIndex Evaluating constraints for table: users [index=0 column=8 term=1 usable=1]
osquery planner: xBestIndex Recording constraint set for table: users [cost=1000000.000000 size=0 idx=1]
osquery planner: xOpen Opening cursor (0) for table: managed_policies
osquery planner: xOpen Opening cursor (1) for table: users
osquery planner: xFilter Filtering called for table: managed_policies [constraint_count=1 argc=0 idx=0]
osquery planner: Scanning rows for cursor (0)
osquery planner: xFilter managed_policies generate returned row count:37
osquery planner: xFilter Filtering called for table: users [constraint_count=1 argc=0 idx=1]
osquery planner: Scanning rows for cursor (1)
osquery planner: xFilter users generate returned row count:113
osquery planner: xFilter Filtering called for table: users [constraint_count=1 argc=0 idx=1]
osquery planner: Scanning rows for cursor (1)
osquery planner: xFilter users generate returned row count:113
osquery planner: xFilter Filtering called for table: users [constraint_count=1 argc=0 idx=1]
osquery planner: Scanning rows for cursor (1)
osquery planner: xFilter users generate returned row count:113
osquery planner: xFilter Filtering called for table: users [constraint_count=1 argc=0 idx=1]
osquery planner: Scanning rows for cursor (1)

[snip]
```

Whereas on this branch, the same command results in:

```
osquery planner: xBestIndex Evaluating constraints for table: managed_policies [index=0 column=1 term=0 usable=1]
osquery planner: xBestIndex Recording constraint set for table: managed_policies [cost=1000000.000000 size=0 idx=0]
osquery planner: xBestIndex Evaluating constraints for table: users [index=0 column=8 term=1 usable=1]
osquery planner: xBestIndex Adding index constraint for table: users [column=uuid arg_index=1 op=2]
osquery planner: xBestIndex Recording constraint set for table: users [cost=1.000000 size=1 idx=1]
osquery planner: xBestIndex Evaluating constraints for table: users [index=0 column=8 term=1 usable=0]
osquery planner: xBestIndex Recording constraint set for table: users [cost=1000000.000000 size=0 idx=2]
osquery planner: xOpen Opening cursor (0) for table: managed_policies
osquery planner: xOpen Opening cursor (1) for table: users
osquery planner: xFilter Filtering called for table: managed_policies [constraint_count=1 argc=0 idx=0]
osquery planner: Scanning rows for cursor (0)
osquery planner: xFilter managed_policies generate returned row count:37
osquery planner: xFilter Filtering called for table: users [constraint_count=2 argc=1 idx=1]
osquery planner: Scanning rows for cursor (1)
osquery planner: xFilter users generate returned row count:113
osquery planner: xFilter Filtering called for table: users [constraint_count=2 argc=1 idx=1]
osquery planner: Scanning rows for cursor (1)
osquery planner: xFilter users generate returned row count:113
osquery planner: xFilter Filtering called for table: users [constraint_count=2 argc=1 idx=1]
osquery planner: Scanning rows for cursor (1)

[snip]
```